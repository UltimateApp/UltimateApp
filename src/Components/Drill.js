/** Represents a the successive positions of the elements in a drill
   Each drill has an attribute "positions" which is a 4D array:
 - positions[elemId][stepId][subStepId][coordinateId]

 Remark1: positions[elemId][0] is the initial position which must be defined for each element
 Remark2: positions[elemId][stepId] is undefined if  element i position does not change between steps stepId-1 and stepId
*/
class Drill {
  constructor(animation) {
    console.log(' posistions are ', JSON.stringify(animation.positions));
    this.positions = animation.positions || [];
    this.ids = animation.ids || [];
    this.texts = animation.texts || [];
  }

  /** Get the position of an element at a given step or return null if its position at step stoppingStep is the last one at step stepId */
  getPositionsAtStep(elemId, stepId, stoppingStep = -1) {
    /* Get the position of the element at step stepId */
    console.log('positions are ', JSON.stringify(this.positions[stepId]));
    let nextPosition = this.positions[stepId][elemId];

    /* If the element does not change its position at step stepId, find its last previous position */
    var stepToCheck = stepId - 1;

    /* While the position of the element at step stepId has not been found  and all the steps after the current one have been checked */
    while ((nextPosition === null || nextPosition === undefined) && stepToCheck !== stoppingStep) {
      nextPosition = this.positions[stepToCheck][elemId];
      stepToCheck -= 1;
    }

    return nextPosition;
  }

  /** Number of steps in the drill */
  stepCount() {
    return this.positions.length;
  }

  /** Number of elements displayed in the drill */
  elemCount() {
    return this.ids.length;
  }

  /** Add an element to the drill */
  addElement(element, initialX, initialY) {
    // Set its initial position
    this.positions[0].push([[initialX, initialY]]);

    // Add an undefined representing the fact that it does not currently move at the other steps
    for (var i = 1; i < this.stepCount(); ++i) this.positions[i].push(undefined);

    // Add its text
    element.props.number == undefined ? this.texts.push('') : this.texts.push(element.props.number);

    // Add its type
    this.ids.push(element.props.id);
  }
}

export default Drill;
