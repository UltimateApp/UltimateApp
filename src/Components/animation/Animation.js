import React from 'react';
import { Easing, Animated, Dimensions, View, Platform } from 'react-native';

import DisplayedElement from './DisplayedElement';
import DisplayedCuts from './DisplayedCuts';
import ProgressBar from './ProgressBar';
import AnimationBackground from './AnimationBackground';

class Animation extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      stepLength: 1000, // Duration of a step in milliseconds
      currentStep: props.currentStep || 0, // Current step displayed on the phone
      animationPlaying: false,
    };

    // Enables to update the current step inside an animation
    if (this.props.currentStepAV !== null && this.props.currentStepAV !== undefined)
      this.currentStepAV = this.props.currentStepAV;
    else this.currentStepAV = new Animated.Value(0);

    this.currentStepAV.addListener((progress) => {
      this.setState({ currentStep: progress.value });
    });

    this.animationWidth = 100;
    this.animationHeight = 100;

    // Distance between the top of the window and the animation area
    this.dTop = this.props.dTop || 0;

    // Distance between the left of the window and the animation area
    if (Platform.OS === 'ios') {
      this.dLeft = 0;
    } else {
      this.dLeft = this.props.dLeft || 0;
    }
  }

  /** Convert a position (x, y) in percentages of the animation area in a position (x2, y2) in pixels of the phone screen
   * x: horizontal position in percentages (=0 left edge, =1 right edge)
   * y: vertical position in percentages (=0 top, =1 bottom)
   * x2: corresponding horizontal position in pixel (=0 if centered)
   * y2: corresponding vertical position in pixel (=0 if centered)
   */
  _positionPercentToPixel = (x, y) => {
    if (Platform.OS === 'ios') {
      return [this.animationWidth * x + 0, this.animationHeight * y + this.dTop];
    } else {
      return [this.animationWidth * x + this.dLeft, this.animationHeight * y + this.dTop];
    }
  };

  componentDidMount() {
    var { height, width } = Dimensions.get('window');

    this.animationWidth = width * this.props.widthRatio;

    this.animationHeight = height * this.props.heightRatio;
    if (this.props.editable) this.props.onDimensionSet(this.animationHeight, this.animationWidth);
  }

  componentDidUpdate(prevProps) {
    if (!this.props.editable && prevProps.animation !== this.props.animation) {
      this.currentStepAV.setValue(0);
    }
  }

  playAnimation = () => {
    this.setState(
      {
        animationPlaying: true,
      },
      () => {
        var sequence = [];

        /* Move instantly to the initial position */
        sequence.push(
          Animated.timing(this.currentStepAV, {
            toValue: 0,
            duration: 0,
            easing: Easing.linear,
            key: 0,
            useNativeDriver: false,
          }),
        );

        sequence.push(
          Animated.timing(this.currentStepAV, {
            toValue: this.props.animation.stepCount() - 1,
            duration: this.state.stepLength * (this.props.animation.stepCount() - 1),
            easing: Easing.linear,
            key: 1,
            useNativeDriver: false,
          }),
        );

        Animated.sequence(sequence).start(() =>
          this.setState({ animationPlaying: false, currentStep: this.props.animation.stepCount() - 1 }),
        );
      },
    );
  };

  playStep = (stepId) => {
    this.setState(
      {
        animationPlaying: true,
      },
      () => {
        stepId = Math.max(0, Math.round(stepId));

        var sequence = [];

        /* Move instantly to the previous step */
        sequence.push(
          Animated.timing(this.currentStepAV, {
            toValue: Math.max(0, stepId - 1),
            duration: 0,
            easing: Easing.linear,
            key: 0,
            useNativeDriver: false,
          }),
        );

        /* If we do not move to the first step, animate the move from the previous step to the current step */
        if (stepId > 0) {
          sequence.push(
            Animated.timing(this.currentStepAV, {
              toValue: stepId,
              duration: this.state.stepLength,
              easing: Easing.linear,
              key: 1,
              useNativeDriver: false,
            }),
          );
        }

        Animated.sequence(sequence).start(() => this.setState({ animationPlaying: false, currentStep: stepId }));
      },
    );
  };

  render() {
    return (
      <View>
        <View
          ref={(ref) => {
            this.marker = ref;
          }}
          onLayout={({ nativeEvent }) => {
            if (this.props.editable) {
              this.marker.measure((x, y, width, height, pageX, pageY) => {
                // On iOS, when the left margin is = 0, pageX can be equal to the whole width instead of 0
                if (pageX > 0.99 * width) pageX = 0;

                if (pageY !== undefined) this.dTop = pageY;

                if (pageX !== undefined) this.dLeft = pageX;
                if (Platform.OS === 'ios') {
                  pageX = 0;
                }

                this.props.onTopMarginSet(this.dTop);
              });
            }
          }}
          style={{ height: this.animationHeight, width: this.animationWidth }}
        >
          {this.animationWidth && (
            <AnimationBackground
              animationWidth={this.animationWidth}
              animationHeight={this.animationHeight}
              background={this.props.animation.background}
            />
          )}

          {this.props.editable && !this.state.animationPlaying && (
            <DisplayedCuts
              step={this.state.currentStep}
              positionPercentToPixel={this._positionPercentToPixel}
              animation={this.props.animation}
              onMoveEnd={this.props.onCutMove}
              topMargin={this.dTop}
            />
          )}
          {this.props.animation.ids.map((id, index) => (
            <DisplayedElement
              type={id}
              number={this.props.animation.texts[index]}
              key={index}
              eId={index}
              editable={this.props.editable}
              onMoveStart={this.props.onMoveStart}
              onMoveEnd={this.props.onElementMoveEnd}
              topMargin={this.dTop}
              animationWidth={this.animationWidth}
              animationHeight={this.animationHeight}
              animation={this.props.animation}
              currentStepAV={this.currentStepAV}
              positionPercentToPixel={this._positionPercentToPixel}
            />
          ))}
        </View>
        <View>
          <ProgressBar
            readonly={!this.props.editable}
            animationWidth={this.animationWidth}
            animationHeight={this.animationHeight}
            stepCount={this.props.animation.stepCount()}
            currentStepAV={this.currentStepAV}
            goToStep={this.playStep}
            playAnimation={this.playAnimation}
            onStepAdded={this.props.onStepAdded}
            onStepRemoved={this.props.onStepRemoved}
          />
        </View>
      </View>
    );
  }
}

export default Animation;
