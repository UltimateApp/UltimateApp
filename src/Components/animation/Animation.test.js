import React from 'react';
import renderer from 'react-test-renderer';

import animationSquare from '../../Fixtures/Animation/AnimationSquare';
import Drill from './Drill';

import Animation from './Animation';

describe('<Animation />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Animation animation={new Drill(animationSquare)} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
