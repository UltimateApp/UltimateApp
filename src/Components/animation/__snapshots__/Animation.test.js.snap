// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Animation /> renders correctly 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  >
    <View
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#232b2b",
              "borderWidth": 1,
              "position": "absolute",
            },
            Object {
              "height": 85,
              "left": 95,
              "top": 5,
              "width": 0.425,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#232b2b",
              "borderWidth": 1,
              "position": "absolute",
            },
            Object {
              "height": 85,
              "left": 5,
              "top": 5,
              "width": 0.425,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#232b2b",
              "borderWidth": 1,
              "position": "absolute",
            },
            Object {
              "height": 0.425,
              "left": 5,
              "top": 5,
              "width": 90,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#232b2b",
              "borderWidth": 1,
              "position": "absolute",
            },
            Object {
              "height": 0.425,
              "left": 5,
              "top": 39,
              "width": 90,
            },
          ]
        }
      />
      <Text />
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#feae19",
          "borderBottomWidth": 4.166666666666667,
          "borderColor": "transparent",
          "borderLeftWidth": 2.0833333333333335,
          "borderRightWidth": 2.0833333333333335,
          "borderStyle": "solid",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 28.958333333333332,
            },
            Object {
              "translateY": 28.958333333333332,
            },
          ],
        }
      }
    >
      <Text
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      >
        
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#feae19",
          "borderBottomWidth": 4.166666666666667,
          "borderColor": "transparent",
          "borderLeftWidth": 2.0833333333333335,
          "borderRightWidth": 2.0833333333333335,
          "borderStyle": "solid",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 28.958333333333332,
            },
            Object {
              "translateY": 58.958333333333336,
            },
          ],
        }
      }
    >
      <Text
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      >
        
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#feae19",
          "borderBottomWidth": 4.166666666666667,
          "borderColor": "transparent",
          "borderLeftWidth": 2.0833333333333335,
          "borderRightWidth": 2.0833333333333335,
          "borderStyle": "solid",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 58.958333333333336,
            },
            Object {
              "translateY": 28.958333333333332,
            },
          ],
        }
      }
    >
      <Text
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      >
        
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#feae19",
          "borderBottomWidth": 4.166666666666667,
          "borderColor": "transparent",
          "borderLeftWidth": 2.0833333333333335,
          "borderRightWidth": 2.0833333333333335,
          "borderStyle": "solid",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 58.958333333333336,
            },
            Object {
              "translateY": 58.958333333333336,
            },
          ],
        }
      }
    >
      <Text
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      >
        
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#feae19",
          "borderBottomWidth": 4.166666666666667,
          "borderColor": "transparent",
          "borderLeftWidth": 2.0833333333333335,
          "borderRightWidth": 2.0833333333333335,
          "borderStyle": "solid",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 43.958333333333336,
            },
            Object {
              "translateY": 18.958333333333332,
            },
          ],
        }
      }
    >
      <Text
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      >
        
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#feae19",
          "borderBottomWidth": 4.166666666666667,
          "borderColor": "transparent",
          "borderLeftWidth": 2.0833333333333335,
          "borderRightWidth": 2.0833333333333335,
          "borderStyle": "solid",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 18.958333333333332,
            },
            Object {
              "translateY": 43.958333333333336,
            },
          ],
        }
      }
    >
      <Text
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      >
        
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#feae19",
          "borderBottomWidth": 4.166666666666667,
          "borderColor": "transparent",
          "borderLeftWidth": 2.0833333333333335,
          "borderRightWidth": 2.0833333333333335,
          "borderStyle": "solid",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 43.958333333333336,
            },
            Object {
              "translateY": 68.95833333333333,
            },
          ],
        }
      }
    >
      <Text
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      >
        
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#feae19",
          "borderBottomWidth": 4.166666666666667,
          "borderColor": "transparent",
          "borderLeftWidth": 2.0833333333333335,
          "borderRightWidth": 2.0833333333333335,
          "borderStyle": "solid",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 68.95833333333333,
            },
            Object {
              "translateY": 43.958333333333336,
            },
          ],
        }
      }
    >
      <Text
        style={
          Object {
            "height": 0,
            "width": 0,
          }
        }
      >
        
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2a9e91",
          "borderRadius": 8.333333333333334,
          "height": 8.333333333333334,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 25.833333333333332,
            },
            Object {
              "translateY": 25.833333333333332,
            },
          ],
          "width": 8.333333333333334,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        1
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2a9e91",
          "borderRadius": 8.333333333333334,
          "height": 8.333333333333334,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 11.833333333333332,
            },
            Object {
              "translateY": 25.833333333333332,
            },
          ],
          "width": 8.333333333333334,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        2
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2a9e91",
          "borderRadius": 8.333333333333334,
          "height": 8.333333333333334,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 55.833333333333336,
            },
            Object {
              "translateY": 25.833333333333332,
            },
          ],
          "width": 8.333333333333334,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        3
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2a9e91",
          "borderRadius": 8.333333333333334,
          "height": 8.333333333333334,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 55.833333333333336,
            },
            Object {
              "translateY": 55.833333333333336,
            },
          ],
          "width": 8.333333333333334,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        4
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2a9e91",
          "borderRadius": 8.333333333333334,
          "height": 8.333333333333334,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 25.833333333333332,
            },
            Object {
              "translateY": 55.833333333333336,
            },
          ],
          "width": 8.333333333333334,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        5
      </Text>
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 5.555555555555556,
          "borderWidth": 0.5555555555555556,
          "height": 5.555555555555556,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAlignVertical": "center",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 33.22222222222222,
            },
            Object {
              "translateY": 33.22222222222222,
            },
          ],
          "width": 5.555555555555556,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#f1f1f1",
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "gray",
                "height": 4,
                "position": "absolute",
                "top": 5.5,
              },
              Object {
                "height": 10,
                "top": 12.5,
              },
              Object {
                "width": 54.166666666666664,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "black",
              "height": 10,
              "position": "absolute",
              "top": 12.5,
              "width": 0,
            }
          }
        />
        <View
          left={0}
          style={
            Object {
              "backgroundColor": "white",
              "height": "100%",
              "position": "absolute",
              "width": 7.5,
            }
          }
        />
        <View
          left={10.833333333333334}
          style={
            Object {
              "backgroundColor": "white",
              "height": "100%",
              "position": "absolute",
              "width": 7.5,
            }
          }
        />
        <View
          left={21.666666666666668}
          style={
            Object {
              "backgroundColor": "white",
              "height": "100%",
              "position": "absolute",
              "width": 7.5,
            }
          }
        />
        <View
          left={32.5}
          style={
            Object {
              "backgroundColor": "white",
              "height": "100%",
              "position": "absolute",
              "width": 7.5,
            }
          }
        />
        <View
          left={43.333333333333336}
          style={
            Object {
              "backgroundColor": "white",
              "height": "100%",
              "position": "absolute",
              "width": 7.5,
            }
          }
        />
        <View
          left={54.16666666666667}
          style={
            Object {
              "backgroundColor": "white",
              "height": "100%",
              "position": "absolute",
              "width": 7.5,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "width": 0,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "width": 10.833333333333334,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 10.833333333333334,
              "opacity": 1,
              "position": "absolute",
              "width": 10.833333333333334,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 21.666666666666664,
              "opacity": 1,
              "position": "absolute",
              "width": 10.833333333333334,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 32.5,
              "opacity": 1,
              "position": "absolute",
              "width": 10.833333333333334,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 43.333333333333336,
              "opacity": 1,
              "position": "absolute",
              "width": 10.833333333333334,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "paddingHorizontal": 10,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
</View>
`;
